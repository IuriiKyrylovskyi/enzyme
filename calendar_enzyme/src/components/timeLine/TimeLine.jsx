import React, { Component } from 'react';
import './timeline.scss';

class TimeLine extends Component {
  state = {
    date: new Date(),
    isUpdate: true,
  }

  componentDidMount() {
    if (this.isCurrentDate()) {

      const timeSecDiff = 60 * 1000 - (new Date()).getMilliseconds();

      setTimeout(() => this.interval, timeSecDiff);

      this.interval = setInterval(() => {
        this.setState({
          date: new Date(),
          isUpdate: true,
        })
      }, 1000)
    }
  }

  componentWillUnmount() {
    this.setState({
      ...this.state,
      isUpdate: false
    })
    clearInterval(this.interval)
  }

  isCurrentDate = () => {
    const weekStartDate = this.props.weekStartDate.getTime();
    const weekEndDate = weekStartDate + 7 * 24 * 60 * 60* 1000;
    const nowTime = (new Date()).getTime();
 
    return (nowTime < weekEndDate && nowTime > weekStartDate);
  } 

  render() {
    console.log('render timeLine');
    const mins = this.state.date.getMinutes();
  
    return (
      this.state.isUpdate &&
      <div
        className="time-line"
        style={{top: mins}}
      >
      </div>
    )
  }
}

export default TimeLine;