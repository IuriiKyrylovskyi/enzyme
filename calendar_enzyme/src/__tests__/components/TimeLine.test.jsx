import { shallow } from 'enzyme'
import React from 'react'
import TimeLine from '../../components/timeLine/TimeLine'

describe("<TimeLine />", () => {
  const wrappedComponent = shallow(<TimeLine />)

  it('should render component', () => {
    expect(wrappedComponent).toMatchSnapshot()
  })
});