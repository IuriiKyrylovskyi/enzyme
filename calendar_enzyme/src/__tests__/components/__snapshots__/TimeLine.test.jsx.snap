// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeLine /> should render component 1`] = `
<div
  className="time-line"
  style={
    Object {
      "top": 16,
    }
  }
/>
`;
